### PR TITLE
Samples: Bluetooth: Mesh: fix common mesh adv data

### PR DESCRIPTION
**Pull request by m-alperen-sener**
SMP service UUID must be presented in adv data.

### CODE REVIEW

**Comment by omkar3141** [comment]
I don't think this is a good idea. You can't assume that the external client will always try to do initiate scan-response from the peripheral.

**Comment by m-alperen-sener** [comment]
> I don't think this is a good idea. You can't assume that the external client will always try to do initiate scan-response from the peripheral.

Can you elaborate please, I don't understand the point.

File: samples/bluetooth/mesh/common/smp_bt.c [hunk 1]
```diff
@@ -34,12 +34,11 @@ static struct bt_le_adv_param adv_params = {
 
 static const struct bt_data ad[] = {
 	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
-	BT_DATA(BT_DATA_NAME_COMPLETE, CONFIG_BT_DEVICE_NAME, sizeof(CONFIG_BT_DEVICE_NAME) - 1),
+	BT_DATA_BYTES(BT_DATA_UUID128_ALL, 0x84, 0xaa, 0x60, 0x74, 0x52, 0x8a, 0x8b, 0x86, 0xd3,
```
Comment (alxelax) [inline samples/bluetooth/mesh/common/smp_bt.c:37]: Will it show device name in the list of adv devices? I guess passive scanning is used for this and device name in form of AD.
Could you check it?
Scan Response is available only for active scanning.

Comment (m-alperen-sener) [reply samples/bluetooth/mesh/common/smp_bt.c:37]: yes that is true; I ll update the PR so in passive scanning the name also appears. Thanks for warning.

Comment (m-alperen-sener) [reply samples/bluetooth/mesh/common/smp_bt.c:37]: I have reverted that part and made sure that UUID is in adv data.

**Review by alxelax — APPROVED** [review]

**Comment by omkar3141** [comment]
Is that UUID a mesh node UUID or SMP Service UUID? If it is mesh node UUID, it does not seem right to add this as `BT_DATA_UUID128_ALL` (this is intended for services), it should be added as `BT_DATA_MANUFACTURER_DATA`.

**Comment by ludvigsj** [comment]
> Is that UUID a mesh node UUID or SMP Service UUID? If it is mesh node UUID, it does not seem right to add this as `BT_DATA_UUID128_ALL` (this is intended for services), it should be added as `BT_DATA_MANUFACTURER_DATA`.

It is the service UUID for SMP, see this page https://docs.zephyrproject.org/latest/services/device_mgmt/smp_transport.html

**Comment by omkar3141** [comment]
> It is the service UUID for SMP, see this page https://docs.zephyrproject.org/latest/services/device_mgmt/smp_transport.html

Then the mystery is solved!

**Review by omkar3141 — APPROVED** [review]
It will nice to add one liner comment there to make it tiny bit self explanatory.

**Pull request merged**